### PR TITLE
cmd/install: warn on cask install when already installed

### PR DESCRIPTION
**Pull request by apainintheneck**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Mentioned in #15295.

Basically we want to warn here instead of erroring out to avoid breaking people's scripts. This was the default behavior before #15273 got merged in to refactor the `cask/cmd/install` logic into `cmd/install`.

### CODE REVIEW

File: Library/Homebrew/cmd/install.rb [hunk 1]
```diff
@@ -197,10 +197,10 @@ def self.install
     begin
       formulae, casks = args.named.to_formulae_and_casks
                             .partition { |formula_or_cask| formula_or_cask.is_a?(Formula) }
-    rescue FormulaOrCaskUnavailableError, Cask::CaskUnavailableError => e
+    rescue FormulaOrCaskUnavailableError, Cask::CaskUnavailableError
       retry if Tap.install_default_cask_tap_if_necessary(force: args.cask?)
 
-      raise e
+      raise
```
Comment (apainintheneck) [inline Library/Homebrew/cmd/install.rb:203]: This is needed to avoid a typechecking error.

```
/u/l/Homebrew (warn-on-cask-install-when-already-installed|✔) $ brew tc
cmd/install.rb:240: Changing the type of a variable in a loop is not permitted https://srb.help/7001
     240 |      rescue Cask::CaskAlreadyInstalledError => e
                                                          ^
  Existing variable has type: NilClass
  Attempting to change type to: T.untyped

  Original type from:
    cmd/install.rb:198:
     198 |      formulae, casks = args.named.to_formulae_and_casks
     199 |                            .partition { |formula_or_cask| formula_or_cask.is_a?(Formula) }
     200 |    rescue FormulaOrCaskUnavailableError, Cask::CaskUnavailableError => e
     201 |      retry if Tap.install_default_cask_tap_if_necessary(force: args.cask?)
     202 |
     203 |      raise e
    cmd/install.rb:168: Possibly uninitialized (NilClass) in:
     168 |  def self.install
            ^^^^^^^^^^^^^^^^
Errors: 1
Check https://docs.brew.sh/Typechecking for more information on how to resolve these errors.
```

It should behave same way though.

File: Library/Homebrew/cmd/install.rb [hunk 2]
```diff
@@ -237,6 +237,8 @@ def self.install
                             skip_cask_deps: args.skip_cask_deps?,
                             quarantine:     args.quarantine?,
                             quiet:          args.quiet?).install
+      rescue Cask::CaskAlreadyInstalledError => e
+        opoo e.message
```
Comment (apainintheneck) [inline Library/Homebrew/cmd/install.rb:241]: Previously it worked like this:

https://github.com/Homebrew/brew/blob/cba118533b83ba1587cf65da0142f58523d210e7/Library/Homebrew/cask/cmd/install.rb#L108-L112

**Review by MikeMcQuaid — APPROVED** [review]
Thanks again @apainintheneck!
